### PR TITLE
Source debug ID insertion fixes.

### PR DESCRIPTION
**Pull request by perf2711**
This PR fixes two issues:

* When the source has a shebang on the first line, the debug ID snippet was inserted before it and not after it
* When the source had trailing whitespaces, the debug ID comment was inserted after them, and not before them

### CODE REVIEW

File: tools/sourcemap-tools/src/helpers/stringHelpers.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ * const value = 'def';
+ * const appended = appendBeforeWhitespaces(str, value);
+ */
+export function appendBeforeWhitespaces(str: string, value: string) {
+    const whitespaces = str.match(/\s*$/)?.[0];
+    if (!whitespaces) {
+        return str + value;
+    }
+
+    return str.substring(0, whitespaces.length) + value + whitespaces;
```
Comment (adamcronin42) [inline tools/sourcemap-tools/src/helpers/stringHelpers.ts:16]: I may not be understanding this correctly, but do we want to use `whitespaces.length` here? Wouldn't we want str without the whitespaces + value + whitespaces?

Comment (perf2711) [reply tools/sourcemap-tools/src/helpers/stringHelpers.ts]: Yeah, I screwed that up. Thanks for the heads-up.

I've added some unit tests.

**Review by adamcronin42 — APPROVED** [review]
Looks good!

**Review by konraddysput — APPROVED** [review]

File: tools/sourcemap-tools/src/helpers/stringHelpers.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/**
+ * Appends `value` to `str` before trailing whitespaces in `str`.
+ * @param str String to append to.
+ * @param value String to append.
+ * @example
+ * const str = 'abc\n\n';
+ * const value = 'def';
+ * const appended = appendBeforeWhitespaces(str, value);
```
[the remark this answers is not in the document]
Comment (perf2711) [reply tools/sourcemap-tools/src/helpers/stringHelpers.ts]: done

**Pull request merged**